src/site/antora/modules/ROOT/pages/manual/webapp.adoc

```diff
@@ -15,311 +15,626 @@
-            @Override
-            public void run() {
-                final Log4jWebSupport webSupport =
-                    WebLoggerContextUtils.getWebLifeCycle(TestAsyncServlet.this.getServletContext());
-                webSupport.setLoggerContext();
-                // do stuff
-                webSupport.clearLoggerContext();
-            }
-        });
-    }
-}
+[IMPORTANT]
+====
+These two approaches deliver similar results for log events generates by the web applications themselves or the libraries bundled in the WAR or EAR archive.
+
+Differences between these approaches appear in the handling of **shared** libraries.
+There are two kinds of shared libraries:
+
+. Shared libraries that use **static** `Logger` fields.
+These libraries will always use the same logger context, which will not be one of the per-application contexts.
++
+This kind includes all the shared libraries, which were not written with Jakarta EE in mind.
+
+. Shared libraries that use **instance** `Logger` fields.
+These libraries will use the logger context associated to the web application that uses them.
++
+Application server implementation usually use instance `Logger` fields.
+
+Since the first kind of libraries is more common, counterintuitively the <<single-context>> approach will usually give better results than the <<multiple-contexts>> approach.
+====
+
+[#single-context]
+==== Single logger context
+
+By default, Log4j Core creates a separate logger context per classloader.
+In order to use a single logger context you need to set the
+xref:manual/systemproperties.adoc#log4j2.contextSelector[`log4j2.contextSelector`]
+system property to:
+
+* either
+link:../javadoc/log4j-core/org/apache/logging/log4j/core/selector/BasicContextSelector.html[`org.apache.logging.log4j.core.selector.BasicContextSelector`]
+to use synchronous loggers,
+* or
+link:../javadoc/log4j-core/org/apache/logging/log4j/core/async/BasicAsyncLoggerContextSelector.html[`org.apache.logging.log4j.core.async.BasicAsyncLoggerContextSelector`]
+to use asynchronous loggers.
+
+In this approach you must use xref:manual/lookups.adoc[lookups] to register the application that generated a log event.
+The most useful lookups in this case are:
+
+Web lookup::
+It does not require any setup, but it is available only after `Log4jServletFilter` has been executed.
+Some log events pertinent to a web application can be unmarked.
+See xref:manual/lookups.adoc#WebLookup[web lookup] for more information.
+
+JNDI lookup::
+It covers a larger part of the handling of a request, but it requires additional setup to export the name of the application via JNDI.
+See xref:manual/lookups.adoc#JndiLookup[JNDI lookup] for more information.
+
+When using a single logger context you choose between:
+
+* logging all events to a single appender.
+You'll need a tool like Elastic Search to separate log events.
+We recommend the usage of a structured layout like
+xref:manual/json-template-layout.adoc[]
+in this case:
```

Comment:
```suggestion
We strongly recommend using a structured layout (e.g., xref:manual/json-template-layout.adoc[]) with an additional field capturing the Servlet context name.
This would allow separation of application logs by filtering on the context name.
Following example demonstrates this scheme using a Socket Appender writing to Elasticsearch:
```

The example Log4j configuration files need to be adapted too.